[Reporting] Document steps to grant users reporting access under Basic (#127513)
elastic/kibana: docs/setup/configuring-reporting.asciidoc

<titleabbrev>Configure reporting</titleabbrev>
++++

> For security, you will need to grant users access to the {report-features}, and secure the reporting endpoints

Reviewer: ```suggestion
For security, you grant users access to the {report-features} and secure the reporting endpoints
```

> with TLS/SSL encryption. Additionally, you may need to install graphical packages into the operating system in

Reviewer: ```suggestion
with TLS/SSL encryption. Additionally, you can install graphical packages into the operating system
```

> order for the {kib} server to have screenshotting capability.

Reviewer: ```suggestion
to enable the {kib} server to have screenshotting capabilities.
```

.. Click *Customize*, then click *Analytics*.

> .. Next each application listed, click *All* or customize the privileges more granularly by clicking *Read* and enabling the *Customize sub-feature privileges* checkbox.

Reviewer: ```suggestion
.. For each application, select *All*, or to customize the privileges, select *Read* and *Customize sub-feature privileges*.
```

+

> NOTE: If you have a *Basic* license, sub-feature privileges are not available for your subscription. Refer to the screenshot in <<grant-user-access-basic>>.

Reviewer: ```suggestion
NOTE: If you have a Basic license, sub-feature privileges are unavailable. For details, check out <<grant-user-access-basic>>.
```

[role="screenshot"]
image::user/reporting/images/kibana-privileges-with-reporting.png["Kibana privileges with Reporting options, Gold or higher license"]
+

> NOTE: If *Reporting* options for application features are not available, and the cluster has higher than a *Basic* license, contact your administrator, or <<reporting-advanced-settings,check that xpack.reporting.roles.enabled is set to false in kibana.yml>>.

Reviewer: ```suggestion
NOTE: If the *Reporting* options for application features are unavailable, and the cluster license is higher than Basic, contact your administrator, or <<reporting-advanced-settings,check that `xpack.reporting.roles.enabled` is set to `false` in kibana.yml>>.
```

[float]
[[reporting-roles-user-api]]
==== Grant access with the role API

> With <<grant-user-access, {kib} application privileges>> enabled in Reporting, you can also use the {ref}/security-api-put-role.html[role API] to grant access to the {report-features}, using *All* privileges, or granular sub-feature privileges.

Reviewer: ```suggestion
With <<grant-user-access, {kib} application privileges>> enabled in Reporting, you can also use the {ref}/security-api-put-role.html[role API] to grant access to the {report-features}, using *All* privileges, or sub-feature privileges.
```

> NOTE: If you have a *Basic* license, sub-feature privileges are not available for your subscription. Refer to the API command to grant *All* privileges shown in <<grant-user-access-basic>>.

Reviewer: ```suggestion
NOTE: If you have a Basic license, sub-feature privileges are unavailable. For details, check out the API command to grant *All* privileges in <<grant-user-access-basic>>.
```

> Grant custom reporting roles to users in combination with other roles that grant read access to the data in {es}, and at least read access in the applications where users can generate reports.

Reviewer: ```suggestion
Grant users custom Reporting roles, other roles that grant read access to the data in {es}, and at least read access in the applications where users can generate reports.
```

<5> Grants access to generate PNG and PDF reports in *Visualize Library*.

[float]
[[grant-user-access-basic]]
=== Grant users access with a Basic license

> With a Basic license and enabled, you are still able to grant users access with custom roles to {report-features} with <<kibana-privileges, {kib} application privileges>>. However, with a Basic license, sub-feature privileges are not available. You will need to create a role <<grant-user-access, as desribed above>>, but select *All* privileges for the applications for which to allow users to create reports.

Reviewer: ```suggestion
With a Basic license, you can grant users access with custom roles to {report-features} with <<kibana-privileges, {kib} application privileges>>. However, with a Basic license, sub-feature privileges are unavailable. <<grant-user-access,Create a role>>, then select *All* privileges for the applications where users can create reports.
```

[role="screenshot"]
image::user/reporting/images/kibana-privileges-with-reporting-basic.png["Kibana privileges with Reporting options, Basic license"]

> You can also use the {ref}/security-api-put-role.html[role API] to grant access to the {report-features}. With

Reviewer: ```suggestion
```

> a *Basic* license, sub-feature application privileges are not available, so the API call must be 

Reviewer: ```suggestion
With a Basic license, sub-feature application privileges are unavailable, but you can use the {ref}/security-api-put-role.html[role API] to grant access to the {report-features}:
```

[source, sh]
---------------------------------------------------------------
POST /_security/role/custom_reporting_user
{
  metadata: {},
  elasticsearch: { cluster: [], indices: [], run_as: [] },
  kibana: [
    {
      spaces": ['*'],
      base": [],
      feature: {
        discover: [ 'all' ] <1>
        dashboard: [ 'all' ] <2>
      }
    }
  ]
}
---------------------------------------------------------------
// CONSOLE

<1> Grants access to generate CSV reports from saved searches in *Discover*.

> <2> Grants access to download CSV from saved search panels in *Dashboard*.

Reviewer: ```suggestion
<2> Grants access to download CSV reports from saved search panels in *Dashboard*.
```
